internal/backend/telemetry/telemetry.go

```diff
@@ -0,0 +1,120 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	api "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"go.uber.org/zap"
+)
+
+func toMetricAttrs(attrs map[string]string) []attribute.KeyValue {
+	var result []attribute.KeyValue
+	for k, v := range attrs {
+		result = append(result, attribute.String(k, v))
+	}
+	return result
+}
+
+type Metric interface {
+	Update(value int64, attrs map[string]string)
+}
+
+type Counter struct {
+	counter api.Int64Counter
+}
+
+type UpDownCounter struct {
+	counter api.Int64UpDownCounter
+}
+
+func (m Counter) Update(value int64, attrs map[string]string) {
+	// REVIEW: should we check/report if we get a negative value
```

Comment:
see my other comment, 
expose a nicer simple API to the user and handle all the ugly stuff here
if you don't want to allow negative value in counter (I think that's a valid idea), don't allow it

Reply:
see my comments above